No type named std::iota and std::random_device

Mavericks, Yosemite and El Capitan give the same error.
http://bot.brew.sh/job/Homebrew%20Science%20Pull%20Requests/4447/version=el_capitan/testReport/junit/brew-test-bot/el_capitan/install_homebrew_science_sailfish/
```
cd /tmp/sailfish20160406-61846-1fmc239/sailfish-0.9.2/build/src && /usr/local/Library/ENV/4.3/clang++    -I/tmp/sailfish20160406-61846-1fmc239/sailfish-0.9.2/tests -I/tmp/sailfish20160406-61846-1fmc239/sailfish-0.9.2/include -I/tmp/sailfish20160406-61846-1fmc239/sailfish-0.9.2/include/eigen3 -I/tmp/sailfish20160406-61846-1fmc239/sailfish-0.9.2/external -I/tmp/sailfish20160406-61846-1fmc239/sailfish-0.9.2/external/cereal/include -I/tmp/sailfish20160406-61846-1fmc239/sailfish-0.9.2/external/install/include -I/tmp/sailfish20160406-61846-1fmc239/sailfish-0.9.2/external/install/include/rapmap -I/tmp/sailfish20160406-61846-1fmc239/sailfish-0.9.2/external/install/include/jellyfish-2.2.3 -I/usr/local/include  -pthread -funroll-loops -fPIC -fomit-frame-pointer -Ofast -DHAVE_ANSI_TERM -DHAVE_SSTREAM -DRAPMAP_SALMON_SUPPORT -Wall -std=c++11 -Wreturn-type -Werror=return-type -undefined dynamic_lookup -stdlib=libc++ -DBOOST_HAS_INT128 -Wno-deprecated-register -pthread -funroll-loops -fPIC -fomit-frame-pointer -Ofast -DHAVE_ANSI_TERM -DHAVE_SSTREAM -DRAPMAP_SALMON_SUPPORT -Wall -std=c++11 -Wreturn-type -Werror=return-type -undefined dynamic_lookup -stdlib=libc++ -DBOOST_HAS_INT128 -Wno-deprecated-register -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk -mmacosx-version-min=10.11   -o CMakeFiles/sailfish_core.dir/SailfishQuantify.cpp.o -c /tmp/sailfish20160406-61846-1fmc239/sailfish-0.9.2/src/SailfishQuantify.cpp
In file included from /tmp/sailfish20160406-61846-1fmc239/sailfish-0.9.2/src/SailfishQuantify.cpp:36:
In file included from /tmp/sailfish20160406-61846-1fmc239/sailfish-0.9.2/include/CollapsedEMOptimizer.hpp:11:
/tmp/sailfish20160406-61846-1fmc239/sailfish-0.9.2/include/ReadExperiment.hpp:162:7: error: no member named 'iota' in namespace 'std'
        std::iota(pos.begin(), pos.end(), 0);
        ~~~~~^
/tmp/sailfish20160406-61846-1fmc239/sailfish-0.9.2/src/SailfishQuantify.cpp:169:8: error: no type named 'random_device' in namespace 'std'
  std::random_device rd;
  ~~~~~^
```

--- a/include/ReadExperiment.hpp
+++ b/include/ReadExperiment.hpp
@@ -22,6 +22,7 @@
 #include <memory>
 #include <fstream>
 #include <atomic>
+#include <numeric>
 
 #include "UtilityFunctions.hpp"
 #include "ReadKmerDist.hpp"

--- a/src/SailfishQuantify.cpp
+++ b/src/SailfishQuantify.cpp
@@ -4,6 +4,7 @@
 #include <iostream>
 #include <cstdlib>
 #include <fstream>
+#include <random>
 #include <vector>
 #include <thread>
 

